### PR TITLE
[RepositoryManager] Limit the amount of concurrent lookups

### PR DESCRIPTION
**Pull request by aciidb0mb3r**

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
Don't know how to test this..

File: Sources/SourceControl/RepositoryManager.swift [hunk 1]
```diff
@@ -221,6 +239,11 @@ public class RepositoryManager {
                 DispatchQueue.global().async {
                     completion(result)
                 }
+                self.maxRequestsCondition.whileLocked {
+                    // We have processed this lookup request, decrement the request count and notify the blocked threads.
+                    self.requestCount -= 1
+                    self.maxRequestsCondition.broadcast()
```
Comment (ddunbar) [inline Sources/SourceControl/RepositoryManager.swift]: This can use `signal` not `broadcast` (since only one waiter will be able to pick up request).

**Comment by abertelrud** [comment]
Another approach would be to use a buffer array, only putting blocks into the concurrent queue when previous ones complete, thus keeping the total number bounded.  That of course makes things fairly complicated, since the completion handler for a block needs to enqueue the next item from the buffer (and mutate the buffer under a lock).  But that's another way if NSOperationQueue isn't available on all platforms.

**Comment by ddunbar** [comment]
@swift-ci please test

**Comment by ddunbar** [comment]
@swift-ci please test

**Comment by aciidb0mb3r** [comment]
@swift-ci please test os x

**Pull request merged**